quinn-proto/src/connection/streams/state.rs

```diff
@@ -605,11 +605,16 @@ impl StreamsState {
     }
 
     pub(crate) fn received_ack_of(&mut self, frame: frame::StreamMeta) {
-        let stream = match self.send.get_mut(&frame.id).and_then(|s| s.as_mut()) {
-            None => return,
-            Some(s) => s,
+        let max_send_data = self.max_send_data(&frame.id);
+        let mut entry = match self.send.entry(frame.id) {
+            hash_map::Entry::Vacant(_) => return,
+            hash_map::Entry::Occupied(e) => e,
         };
 
+        let stream = entry
+            .get_mut()
+            .get_or_insert_with(|| Box::new(Send::new(max_send_data)));
```

Comment:
Because we only call this after sending data on this stream, this closure *should* be unreachable. If we did somehow screw that up, then we might hit an underflow below with unpredictable effects down the line, so perhaps a short-circuit or even an unwrap would be better here?